sensor/kubernetes/listener/resource_event_handler.go

```diff
@@ -136,20 +135,45 @@ func (k *listenerImpl) handleAllEvents() {
 	handle(k.context, roleInformer, dispatchers.ForRBAC(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
 	handle(k.context, clusterRoleInformer, dispatchers.ForRBAC(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
 
+	// For openshift clusters only
 	var osConfigFactory osConfigExtVersions.SharedInformerFactory
 	if k.client.OpenshiftConfig() != nil {
-		if ok, err := clusterOperatorCRDExists(k.client); err != nil {
-			log.Errorf("Error checking for cluster operator CRD: %v", err)
-		} else if !ok {
-			log.Warnf("Skipping cluster operator discovery....")
+		if resourceList, err := serverResourcesForGroup(k.client, osConfigGroupVersion); err != nil {
+			log.Errorf("Error checking API resources for group %q: %v", osConfigGroupVersion, err)
 		} else {
 			osConfigFactory = osConfigExtVersions.NewSharedInformerFactory(k.client.OpenshiftConfig(), noResyncPeriod)
+
+			if resourceExists(resourceList, osClusterOperatorsResourceName) {
+				log.Infof("Initializing %q informer", osClusterOperatorsResourceName)
+				handle(k.context, osConfigFactory.Config().V1().ClusterOperators().Informer(), dispatchers.ForClusterOperators(), k.outputQueue, nil, noDependencyWaitGroup, stopSignal, &eventLock)
+			}
+
+			if env.RegistryMirroringEnabled.BooleanSetting() {
+				if resourceExists(resourceList, osImageDigestMirrorSetsResourceName) {
+					log.Infof("Initializing %q informer", osImageDigestMirrorSetsResourceName)
+					handle(k.context, osConfigFactory.Config().V1().ImageDigestMirrorSets().Informer(), dispatchers.ForRegistryMirrors(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
+				}
+
+				if resourceExists(resourceList, osImageTagMirrorSetsResourceName) {
+					log.Infof("Initializing %q informer", osImageTagMirrorSetsResourceName)
+					handle(k.context, osConfigFactory.Config().V1().ImageTagMirrorSets().Informer(), dispatchers.ForRegistryMirrors(), k.outputQueue, &syncingResources, noDependencyWaitGroup, stopSignal, &eventLock)
+				}
+			}
 		}
 	}
-	// For openshift clusters only
-	if osConfigFactory != nil {
-		handle(k.context, osConfigFactory.Config().V1().ClusterOperators().Informer(), dispatchers.ForClusterOperators(),
-			k.outputQueue, nil, noDependencyWaitGroup, stopSignal, &eventLock)
+
+	var osOperatorFactory osOperatorExtVersions.SharedInformerFactory
+	if k.client.OpenshiftOperator() != nil && env.RegistryMirroringEnabled.BooleanSetting() {
+		if resourceList, err := serverResourcesForGroup(k.client, osOperatorAlphaGroupVersion); err != nil {
+			log.Errorf("Error checking API resources for group %q: %v", osOperatorAlphaGroupVersion, err)
+		} else {
+			osOperatorFactory = osOperatorExtVersions.NewSharedInformerFactory(k.client.OpenshiftOperator(), noResyncPeriod)
```

Comment:
does this still need to be set even if the `ImageContentSourcePolicy` is not defined?

Reply:
No it does not. I reviewed the sources for the factory and if no informers have been created using it then `Start()` is a NoOp.

I considered putting it inside `resourceExists` but kept it outside to keep things consistent with the other usages.

I'll move it inside, suppose thats will save on tiny bit of mem during initialization for this condition.

Reply:
Happy to move it for this instance if that's the preference.